test/output_stream_test.rb

```diff
@@ -32,6 +34,15 @@ def test_write_buffer
     assert_test_zip_contents(TEST_ZIP)
   end
 
+  def test_write_buffer_binmode
+    io = ::StringIO.new('')
+    buffer = ::Zip::OutputStream.write_buffer(io) do |zos|
+      zos.comment = TEST_ZIP.comment
+      write_test_zip(zos)
+    end
+    assert buffer.external_encoding.name === ASCII8BIT
```

Comment:
Ruby has a constant for this encoding already:
```suggestion
    assert_equal Encoding::ASCII_8BIT, buffer.external_encoding
```